devices/debian_wifi.py

```diff
@@ -38,51 +47,77 @@ def wpa_connect(self, ssid_name, password):
          else:
             return None
 
-    def wlan_connectivity(self):
+    def wlan_connectivity(self, iface_wlan="wlan1", login_pwd=None):
         '''Connection state verify'''
-        self.sendline("iw %s link" % self.iface_wlan)
-        self.expect(self.prompt)
-        conn_state = self.before
-        match = re.search('Connected',conn_state)
-        if match:
-            return match.group(0)
+        self.sendline("iw %s link" % iface_wlan)
+        try:
+            wlan_connect = self.expect('Connected', timeout=5)
+            self.expect(self.prompt)
+        except:
+            self.sendline("sudo iw %s link" % iface_wlan)
+            idx = self.expect(["password for .*:"] + self.prompt,timeout=20)
+            if idx == 0:
+                self.sendline(login_pwd)
+                wlan_connect = self.expect('Connected', timeout=5)
+                self.expect(self.prompt)
+        if wlan_connect == 0:
+            return "Connected"
         else:
             return None
 
-    def disconnect_wpa(self):
+    def disconnect_wpa(self, login_pwd=None):
```

Comment:
Why the added arg? Do we plan to use it?

Reply:
Replaced to take it from cbn.json